### ISSUE TITLE
Adding support for GeoJSON format.

### ISSUE THREAD

**Issue by MyGIS** [AR-js-org/AR.js#221]
**Do you want to request a *feature* or report a *bug*?**

Is it possible to consider adding support for GeoJSON basic geometries , specifically "Polylines and Polygons" ?

**What is the current behavior?**

**If the current behavior is a bug, please provide the steps to reproduce.**

**Please mention other relevant information such as the browser version, Operating System and Device Name**

**What is the expected behavior?**

**If this is a feature request, what is motivation or use case for changing the behavior?**

**Comment by nickw1** [comment]
@MyGis this would probably be best done as an additional plugin, rather than core to AR.js (@nicolocarpignoli do you agree?) Are you thinking of the ability to download GeoJSON data from a server and render that on an AR.js scene? I already have an A-Frame component to do this (see https://github.com/nickw1/aframe-osm-3d) however at the moment it's optimised for my own use case. If there's interest I can work on this to make it a bit more generic.

**Comment by MyGIS** [comment]
Yes, I am thinking to stream a GeoJSON file. I was thinking more towards the OSM Vector Tiles (https://wiki.openstreetmap.org/wiki/Vector_tiles), more specifically the street polyline (or line) and the name of the street. Let me take a look at your project and revert to you. It would be awesome to have GeoJSON integrated with AR.js .

**Comment by nickw1** [comment]
Definitely yes. As it happens my project does use XYZ vector tiling; it downloads your current tile and the 8 surrounding, so I'm sure it could be adapted to fit your needs (and make more generic).

**Comment by MyGIS** [comment]
Roads layers is all what I need for now. Does it support labeling road names?
Please let me know if you require help for testing.

**Comment by nickw1** [comment]
It does not, but this could be added without too much difficulty.
The components should just work as-is, there is some documentation in the README though admittedly this could be expanded.

There are two components: `terrarium-dem` which downloads Terrarium Digital Elevation Model data from a given source. This was provided on AWS by Mapzen (who no longer exist, but the data is still there) for free with no usage limits. However you'll need to write a proxy script to access the data, see for example https://github.com/nickw1/hikar.js/blob/master/proxy.php, as the AWS does not have CORS enabled.

Once the DEM data is loaded, it is then sent on to the `osm3d` component to apply elevation to the OSM data, supplied by the server as GeoJSON. (I'd imagine most location-based AR apps would need to take into account elevation when rendering).

Anyway, have a look at this repo and let me know if there are any other features you'd like, besides road naming, One thing that does need to be added is style customisation, at the moment a hard-coded set of rendering rules are used (e.g. roads in white, paths in green).
